skbio/embedding/_embedding.py

```diff
@@ -86,15 +110,212 @@ def __repr__(self):
         --------
         Protein
         """
-        seq = Sequence(str(self._ids))
-
-        rstr = repr(seq)
-        rstr = rstr.replace("Sequence", "SequenceEmbedding")
-        n_indent = 4  # see Sequence.__repr__
-        indent = " " * n_indent
-        dim = self.embedding.shape[1]
-        rstr = rstr.replace(
-            "has gaps",
-            f"embedding dimension: {dim}\n{indent}has gaps",
+        seq = Sequence(self.sequence)
+        rstr = _repr_helper(
+            repr(seq), "Sequence", "SequenceEmbedding", "embedding",
+            regex_match="length", shape=self.embedding.shape[1]
+        )
+        return rstr
+
+
+class EmbeddingVector(Embedding):
+    r"""Store a vector representation for a biological entity."""
+    def __init__(self, vector, obj, **kwargs):
+        super(EmbeddingVector, self).__init__(vector, obj, **kwargs)
+
+    def __str__(self):
+        return self._ids[0].decode('ascii')
+
+    @property
+    def vector(self):
+        return self.embedding.squeeze()
+
+    @property
+    def embedding(self):
+        r""" The embedding tensor. """
```

Comment:
Is "tensor" the right term here? I thought that a tensor is basically an array that can be stored in GPU.